### ISSUE TITLE
[MLS Spec change] Use a hash function for hash identifiers. (#688)

### ISSUE THREAD

**Issue by raphaelrobert** [openmls/openmls#928]
**Link to the exact changes**
https://github.com/mlswg/mls-protocol/pull/688

**Description of the changes**
> Closes #640.

**Comment by franziskuskiefer** [comment]
@nomaxg is taking a stab at this.

I believe you need to comment here for me to be able to assign it to you @nomaxg.

**Comment by nomaxg** [comment]
Yes, work in progress [here](https://github.com/openmls/openmls/compare/main...nomaxg:openmls:hash-identifiers)
